backend/src/ee/services/dynamic-secret/dynamic-secret-service.ts

```diff
@@ -187,12 +209,22 @@ export const dynamicSecretServiceFactory = ({
     if (!folder)
       throw new NotFoundError({ message: `Folder with path '${path}' in environment '${environmentSlug}' not found` });
 
-    const dynamicSecretCfg = await dynamicSecretDAL.findOne({ name, folderId: folder.id });
+    const dynamicSecretCfg = await dynamicSecretDAL.findOneWithMetadata({ name, folderId: folder.id });
     if (!dynamicSecretCfg) {
       throw new NotFoundError({
         message: `Dynamic secret with name '${name}' in folder '${folder.path}' not found`
       });
     }
```

Comment:
Missing a permission check with the new metadata. This would avoid a user loosing their access by giving non accessible metadata